packages/create-fuels/src/lib/getPackageManager.ts

```diff
@@ -10,39 +8,53 @@ const runnableApplicator =
   (command: string = '') =>
     `${commandPrefix} ${command}`;
 
-export const packageMangers = {
+export const packageMangers: Record<
+  PackageManager,
+  {
+    install: string;
+    run: (command: string) => string;
+    name: PackageManager;
+  }
+> = {
   pnpm: {
     install: 'pnpm install',
     run: runnableApplicator('pnpm'),
+    name: 'pnpm',
   },
   npm: {
     install: 'npm install',
     run: runnableApplicator('npm run'),
+    name: 'npm',
   },
   bun: {
     install: 'bun install',
     run: runnableApplicator('bun run'),
+    name: 'bun',
   },
 } as const;
 
-export const getPackageManager = (opts: ProgramOptions) => {
-  const packageMangerOpts = {
-    pnpm: opts.pnpm,
-    npm: opts.npm,
-    bun: opts.bun,
-  };
+export function getUserPkgManager(): PackageManager {
+  const userAgent = process.env.npm_config_user_agent || '';
 
-  const cliChosenPackageManagerSelected = Object.entries(packageMangerOpts)
```

Comment:
Nit: Would looping over the keys to find the package manager be simpler? Then we could remove the `name` property from `packageMangers` entries.

```ts
  Object.keys(packageMangers).forEach((packageManager) => {
    if (userAgent.startsWith(packageManger)) {
      return packageManger as PackageManager;
    }
  });

  warn(`This package manager is not supported. Using npm instead.`);
  return packageMangers.npm.name;
```